src/components/GrowlNotification/index.js

```diff
@@ -55,65 +43,80 @@ class GrowlNotification extends Component {
-        );
-    }
+    const show = useCallback((text, growlType, growlDuration) => {
+        setBodyText(text);
+        setType(growlType);
+        setDuration(growlDuration);
+    }, []);
 
     /**
      * Animate growl notification
      *
      * @param {Number} val
      */
-    fling(val = INACTIVE_POSITION_Y) {
-        Animated.spring(this.state.translateY, {
+    const fling = useCallback((val = INACTIVE_POSITION_Y) => {
+        Animated.spring(translateY, {
             toValue: val,
             duration: 80,
             useNativeDriver: true,
         }).start();
-    }
+    }, []);
 
-    render() {
-        return (
-            <FlingGestureHandler
-                direction={Directions.UP}
-                onHandlerStateChange={({nativeEvent}) => {
-                    if (nativeEvent.state !== State.ACTIVE) {
-                        return;
-                    }
+    useImperativeHandle(
+        ref,
+        () => ({
+            show,
+        }),
+        [show],
+    );
 
-                    this.fling(INACTIVE_POSITION_Y);
-                }}
-            >
-                <View style={styles.growlNotificationWrapper}>
-                    <GrowlNotificationContainer translateY={this.state.translateY}>
-                        <PressableWithoutFeedback
-                            accessibilityLabel={this.state.bodyText}
-                            onPress={() => this.fling(INACTIVE_POSITION_Y)}
-                        >
-                            <View style={styles.growlNotificationBox}>
-                                <Icon
-                                    src={types[this.state.type].icon}
-                                    fill={types[this.state.type].iconColor}
-                                />
-                                <Text style={styles.growlNotificationText}>{this.state.bodyText}</Text>
-                            </View>
-                        </PressableWithoutFeedback>
-                    </GrowlNotificationContainer>
-                </View>
-            </FlingGestureHandler>
-        );
-    }
+    useEffect(() => {
+        Growl.setIsReady();
+    }, []);
+
+    useEffect(() => {
+        if (!duration) {
+            return;
+        }
+
+        fling(0);
+        setTimeout(() => {
+            fling();
+            setDuration(undefined);
+        }, duration);
+    }, [duration, fling]);
+
+    return (
+        <FlingGestureHandler
+            direction={Directions.UP}
+            onHandlerStateChange={({nativeEvent}) => {
+                if (nativeEvent.state !== State.ACTIVE) {
+                    return;
+                }
+
+                fling();
+            }}
+        >
+            <View style={styles.growlNotificationWrapper}>
+                <GrowlNotificationContainer translateY={translateY}>
+                    <PressableWithoutFeedback
+                        accessibilityLabel={bodyText}
+                        onPress={() => fling()}
```

Comment:
I would do just 
```
onPress={fling}
```
if not the `PressableWithoutFeedback` receives a lambda and not the stable function created by `useCallback` (just in case that component needs it)

Reply:
you are right - it was called with a variable previously, but I removed passing the variable here, so now it can be simplified as you recommend

Reply:
resolved